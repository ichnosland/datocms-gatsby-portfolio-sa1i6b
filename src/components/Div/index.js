/**
*
* Div
*
*/

import styled, { css } from 'styled-components';
import media from 'style/mediainjector';

const Div = styled.div`
  display: ${(props) => props.display ? props.display : 'block'};
  position: ${(props) => props.position ? props.position : 'relative'};
  width: ${(props) => props.width};
  max-width: ${(props) => props.maxWidth ? props.maxWidth : '100%'};
  height: ${(props) => props.height};
  color: ${(props) => props.color ? props.color : 'inherit'};
  text-align: ${(props) => props.align};
  float: ${(props) => props.float ? props.float : 'none'};
  margin: ${(props) => props.margin ? props.margin : '0'};
  padding: ${(props) => props.padding ? props.padding : '0'};
  border: ${(props) => props.border ? props.border : 'none'};
  border-width: ${(props) => props.borderwidth};
  border-style: ${(props) => props.borderstyle};
  border-color: ${(props) => props.bordercolor};
  background-color: ${(props) => props.bgColor ? props.bgColor : 'transparent'};
  border-radius: ${(props) => props.radius};
  overflow: ${(props) => props.overflow ? props.overflow : 'visible'};
  user-select: ${(props) => props.userSelect};
  pointer-events: ${(props) => props.pointerEvents};
  z-index: ${(props) => props.zindex};
  ${(props) => props.nobreak && css`
    page-break-inside: avoid;
  `}
  ${media.print`
    ${({ noPrint }) => noPrint && css`
        display:none;
    `}
  `}
`;

export default Div;
