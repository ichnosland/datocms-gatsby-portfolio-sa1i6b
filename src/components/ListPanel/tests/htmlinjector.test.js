import React from 'react';
import { shallow } from 'enzyme';
import MathJax from 'react-mathjax-preview';

import Div from '../Div';

import HtmlInjector, {
  HtmlBlock,
  Solution,
  HtmlBlockFooter,
} from '../HtmlInjector';

const items = {
  id: '1',
  nome: 'Ciao',
  isHtml: true,
  note: true,
  text: 'html',
  solution: 'soluzione',
  esercizi: 16,
  data: '25/01/2018',
};

describe('<HtmlInjector />', () => {
  it('should render a Div', () => {
    const renderedComponent = shallow(<HtmlInjector {...items} />);
    expect(renderedComponent.find(Div).length).toEqual(1);
    expect(renderedComponent.find(HtmlBlock).length).toEqual(1);
    expect(renderedComponent.find(Solution).length).toEqual(1);
    expect(renderedComponent.find(HtmlBlockFooter).length).toEqual(1);
    expect(renderedComponent.find('span').length).toEqual(2);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should not render HtmlBlock', () => {
    const renderedComponent = shallow(
      <HtmlInjector
        {...{
          ...items,
          text: undefined,
        }}
      />
    );
    expect(renderedComponent.find(Div).length).toEqual(1);
    expect(renderedComponent.find(HtmlBlock).length).toEqual(0);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should not render MathJax when hasLatex flag is set', () => {
    const renderedComponent = shallow(
      <HtmlInjector
        {...{
          ...items,
          hasLatex: true,
        }}
      />
    );
    expect(renderedComponent.find(Div).length).toEqual(1);
    expect(renderedComponent.find(HtmlBlock).length).toEqual(1);
    expect(renderedComponent.find(Solution).length).toEqual(1);
    expect(renderedComponent.find(MathJax).length).toEqual(2);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should not render MathJax when editor.watch flag is set', () => {
    const renderedComponent = shallow(
      <HtmlInjector
        {...{
          ...items,
          editor: {
            show: true,
          },
        }}
      />
    );
    expect(renderedComponent.find(Div).length).toEqual(2);
    expect(renderedComponent.find(HtmlBlock).length).toEqual(0);
    expect(renderedComponent.find(Solution).length).toEqual(1);
    expect(renderedComponent.find(MathJax).length).toEqual(0);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should not render Solution', () => {
    const renderedComponent = shallow(
      <HtmlInjector
        {...{
          ...items,
          solution: undefined,
        }}
      />
    );
    expect(renderedComponent.find(Div).length).toEqual(1);
    expect(renderedComponent.find(Solution).length).toEqual(0);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render HtmlBlockFooter w/ only data', () => {
    const renderedComponent = shallow(
      <HtmlInjector
        {...{
          ...items,
          esercizi: undefined,
        }}
      />
    );
    expect(renderedComponent.find(Div).length).toEqual(1);
    expect(renderedComponent.find(HtmlBlockFooter).length).toEqual(1);
    expect(renderedComponent.find('span').length).toEqual(1);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render HtmlBlockFooter w/ data when esercizi is 0', () => {
    const renderedComponent = shallow(
      <HtmlInjector
        {...{
          ...items,
          esercizi: 0,
        }}
      />
    );
    expect(renderedComponent.find(Div).length).toEqual(1);
    expect(renderedComponent.find(HtmlBlockFooter).length).toEqual(1);
    expect(renderedComponent.find('span').length).toEqual(2);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render HtmlBlockFooter w/ data when esercizi is 1', () => {
    const renderedComponent = shallow(
      <HtmlInjector
        {...{
          ...items,
          esercizi: 1,
        }}
      />
    );
    expect(renderedComponent.find(Div).length).toEqual(1);
    expect(renderedComponent.find(HtmlBlockFooter).length).toEqual(1);
    expect(renderedComponent.find('span').length).toEqual(2);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should not render HtmlBlockFooter w/o data e esercizi', () => {
    const renderedComponent = shallow(
      <HtmlInjector
        {...{
          ...items,
          esercizi: undefined,
          data: undefined,
        }}
      />
    );
    expect(renderedComponent.find(Div).length).toEqual(1);
    expect(renderedComponent.find(HtmlBlockFooter).length).toEqual(0);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<HtmlBlock />', () => {
  it('should render a Div', () => {
    const renderedComponent = shallow(<HtmlBlock />);
    expect(renderedComponent.find(Div).length).toEqual(1);
    expect(renderedComponent).toMatchSnapshot();
  });
  it('should render its css w/ props align', () => {
    const align = 'left';
    const renderedComponent = shallow(<HtmlBlock align={align} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props color', () => {
    const color = 'red';
    const renderedComponent = shallow(<HtmlBlock color={color} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props note', () => {
    const renderedComponent = shallow(<HtmlBlock note />);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<HtmlBlockFooter />', () => {
  it('should render a Div', () => {
    const renderedComponent = shallow(<HtmlBlockFooter />);
    expect(renderedComponent.find(Div).length).toEqual(1);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<Solution />', () => {
  it('should render a HtmlBlock', () => {
    const renderedComponent = shallow(<Solution />);
    expect(renderedComponent.find(HtmlBlock).length).toEqual(1);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props color', () => {
    const color = 'red';
    const renderedComponent = shallow(<Solution color={color} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props bgcolor', () => {
    const bgcolor = 'green';
    const renderedComponent = shallow(<Solution bgcolor={bgcolor} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
