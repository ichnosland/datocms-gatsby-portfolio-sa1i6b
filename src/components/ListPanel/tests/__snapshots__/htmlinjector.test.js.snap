// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HtmlBlock /> should render a Div 1`] = `
.c0 {
  color: #00BBEF;
  background-color: rgb(255,255,255);
  border-top: 1px solid rgb(231,236,236);
  padding: 16px;
}

.c0 .segnaposto-admin {
  display: none;
}

.c0 .consegna__titolo {
  display: block;
  padding-bottom: 20px;
  font-weight: 600;
}

.c0 .consegna__contenuto {
  color: #000;
}

.c0 .opzioni {
  margin-top: 10px;
  padding: 10px;
  background-color: rgba(96,141,212,0.1);
  border: 1px solid rgba(35,56,89,.1);
  border-radius: 2px;
}

.c0 .opzioni__titolo {
  display: block;
  padding-bottom: 10px;
}

.c0 .opzioni__contenuto {
  color: #000;
  display: inline-block;
  font-style: italic;
  padding-left: 6px;
}

.c0 .opzioni__contenuto .radiocheck-wrap {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 0;
}

.c0 .opzioni .radio,
.c0 .opzioni .checkbox {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  margin-right: 10px;
  padding-top: 3px;
}

.c0 .opzioni .radio:after,
.c0 .opzioni .checkbox:after {
  content: "";
  display: block;
  width: 14px;
  height: 14px;
  border: 1px solid #00BBEF;
  background: white;
}

.c0 .opzioni .checkbox:after {
  border-radius: 2px;
}

.c0 .opzioni .radio:after {
  border-radius: 100px;
}

.c0 .consegna__contenuto--esercizio {
  margin-top: 20px;
}

.c0 .completa {
  line-height: 2em;
}

.c0 .completa:after {
  content: "........................................";
  font-weight: 400;
}

.c0 .completa.full--width {
  display: block;
  height: 0;
  border-bottom: 1px dashed;
  margin: 2em 0;
}

.c0 .completa.full--width:after {
  content: "";
}

.c0 .completa.full--width:last-child {
  margin-bottom: 1em;
}

.c0 .mytestitaliano .contesto {
  border-bottom: 1px dotted;
}

.c0 .consegna__blocchi {
  margin-top: 10px;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 4px;
  padding: 0 4px;
  border-radius: 2px;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco span {
  padding: 4px 0;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco.analizza {
  border: 1px solid rgba(0,0,0,.4);
}

.c0 .consegna__blocchi .consegna__blocchi-blocco.analizza .segnaposto {
  border-bottom: 1px dashed rgba(0,0,0,.4);
}

.c0 p:first-child {
  margin-top: 0;
}

.c0 p:last-child {
  margin-bottom: 0;
}

.c0 img {
  max-width: 100%;
}

.c0 table {
  border-collapse: collapse;
  border-spacing: 0;
}

.c0 table th {
  background: rgb(231,236,236);
}

.c0 table td,
.c0 table th {
  padding: 0.62em;
  border: 1px solid rgb(231,236,236);
  vertical-align: middle;
}

.c0 .numero-esercizio {
  float: right;
  background-color: rgb(209,209,209);
  color: rgb(255,255,255);
  padding: 3px 6px;
  border-radius: 3px;
  text-align: right;
}

.c0 .editor {
  border: 1px dashed rgb(251,91,111);
}

.c0 .MathJax {
  font-size: 0.9em;
}

.c0 table th {
  background: rgb(231,236,236);
}

.c0 table td,
.c0 table th {
  padding: 0.62em;
  border: 1px solid rgb(231,236,236);
  vertical-align: middle;
}

.c0 .numero-esercizio {
  float: right;
  background-color: rgb(209,209,209);
  color: rgb(255,255,255);
  padding: 3px 6px;
  border-radius: 3px;
  text-align: right;
}

@media print {
  .c0 {
    border: none;
  }
}

@media print {
  .c0 .opzioni {
    background: none;
  }
}

<Div
  className="c0"
  theme={
    Object {
      "brand": "#00BBEF",
    }
  }
/>
`;

exports[`<HtmlBlock /> should render its css w/ props align 1`] = `
.c0 {
  color: #00BBEF;
  background-color: rgb(255,255,255);
  border-top: 1px solid rgb(231,236,236);
  padding: 16px;
}

.c0 .segnaposto-admin {
  display: none;
}

.c0 .consegna__titolo {
  display: block;
  padding-bottom: 20px;
  font-weight: 600;
}

.c0 .consegna__contenuto {
  color: #000;
}

.c0 .opzioni {
  margin-top: 10px;
  padding: 10px;
  background-color: rgba(96,141,212,0.1);
  border: 1px solid rgba(35,56,89,.1);
  border-radius: 2px;
}

.c0 .opzioni__titolo {
  display: block;
  padding-bottom: 10px;
}

.c0 .opzioni__contenuto {
  color: #000;
  display: inline-block;
  font-style: italic;
  padding-left: 6px;
}

.c0 .opzioni__contenuto .radiocheck-wrap {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 0;
}

.c0 .opzioni .radio,
.c0 .opzioni .checkbox {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  margin-right: 10px;
  padding-top: 3px;
}

.c0 .opzioni .radio:after,
.c0 .opzioni .checkbox:after {
  content: "";
  display: block;
  width: 14px;
  height: 14px;
  border: 1px solid #00BBEF;
  background: white;
}

.c0 .opzioni .checkbox:after {
  border-radius: 2px;
}

.c0 .opzioni .radio:after {
  border-radius: 100px;
}

.c0 .consegna__contenuto--esercizio {
  margin-top: 20px;
}

.c0 .completa {
  line-height: 2em;
}

.c0 .completa:after {
  content: "........................................";
  font-weight: 400;
}

.c0 .completa.full--width {
  display: block;
  height: 0;
  border-bottom: 1px dashed;
  margin: 2em 0;
}

.c0 .completa.full--width:after {
  content: "";
}

.c0 .completa.full--width:last-child {
  margin-bottom: 1em;
}

.c0 .mytestitaliano .contesto {
  border-bottom: 1px dotted;
}

.c0 .consegna__blocchi {
  margin-top: 10px;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 4px;
  padding: 0 4px;
  border-radius: 2px;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco span {
  padding: 4px 0;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco.analizza {
  border: 1px solid rgba(0,0,0,.4);
}

.c0 .consegna__blocchi .consegna__blocchi-blocco.analizza .segnaposto {
  border-bottom: 1px dashed rgba(0,0,0,.4);
}

.c0 p:first-child {
  margin-top: 0;
}

.c0 p:last-child {
  margin-bottom: 0;
}

.c0 img {
  max-width: 100%;
}

.c0 table {
  border-collapse: collapse;
  border-spacing: 0;
}

.c0 table th {
  background: rgb(231,236,236);
}

.c0 table td,
.c0 table th {
  padding: 0.62em;
  border: 1px solid rgb(231,236,236);
  vertical-align: middle;
}

.c0 .numero-esercizio {
  float: right;
  background-color: rgb(209,209,209);
  color: rgb(255,255,255);
  padding: 3px 6px;
  border-radius: 3px;
  text-align: right;
}

.c0 .editor {
  border: 1px dashed rgb(251,91,111);
}

.c0 .MathJax {
  font-size: 0.9em;
}

.c0 table th {
  background: rgb(231,236,236);
}

.c0 table td,
.c0 table th {
  padding: 0.62em;
  border: 1px solid rgb(231,236,236);
  vertical-align: middle;
}

.c0 .numero-esercizio {
  float: right;
  background-color: rgb(209,209,209);
  color: rgb(255,255,255);
  padding: 3px 6px;
  border-radius: 3px;
  text-align: right;
}

@media print {
  .c0 {
    border: none;
  }
}

@media print {
  .c0 .opzioni {
    background: none;
  }
}

<Div
  align="left"
  className="c0"
  theme={
    Object {
      "brand": "#00BBEF",
    }
  }
/>
`;

exports[`<HtmlBlock /> should render its css w/ props color 1`] = `
.c0 {
  color: red;
  background-color: rgb(255,255,255);
  border-top: 1px solid rgb(231,236,236);
  padding: 16px;
}

.c0 .segnaposto-admin {
  display: none;
}

.c0 .consegna__titolo {
  display: block;
  padding-bottom: 20px;
  font-weight: 600;
}

.c0 .consegna__contenuto {
  color: #000;
}

.c0 .opzioni {
  margin-top: 10px;
  padding: 10px;
  background-color: rgba(96,141,212,0.1);
  border: 1px solid rgba(35,56,89,.1);
  border-radius: 2px;
}

.c0 .opzioni__titolo {
  display: block;
  padding-bottom: 10px;
}

.c0 .opzioni__contenuto {
  color: #000;
  display: inline-block;
  font-style: italic;
  padding-left: 6px;
}

.c0 .opzioni__contenuto .radiocheck-wrap {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 0;
}

.c0 .opzioni .radio,
.c0 .opzioni .checkbox {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  margin-right: 10px;
  padding-top: 3px;
}

.c0 .opzioni .radio:after,
.c0 .opzioni .checkbox:after {
  content: "";
  display: block;
  width: 14px;
  height: 14px;
  border: 1px solid #00BBEF;
  background: white;
}

.c0 .opzioni .checkbox:after {
  border-radius: 2px;
}

.c0 .opzioni .radio:after {
  border-radius: 100px;
}

.c0 .consegna__contenuto--esercizio {
  margin-top: 20px;
}

.c0 .completa {
  line-height: 2em;
}

.c0 .completa:after {
  content: "........................................";
  font-weight: 400;
}

.c0 .completa.full--width {
  display: block;
  height: 0;
  border-bottom: 1px dashed;
  margin: 2em 0;
}

.c0 .completa.full--width:after {
  content: "";
}

.c0 .completa.full--width:last-child {
  margin-bottom: 1em;
}

.c0 .mytestitaliano .contesto {
  border-bottom: 1px dotted;
}

.c0 .consegna__blocchi {
  margin-top: 10px;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 4px;
  padding: 0 4px;
  border-radius: 2px;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco span {
  padding: 4px 0;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco.analizza {
  border: 1px solid rgba(0,0,0,.4);
}

.c0 .consegna__blocchi .consegna__blocchi-blocco.analizza .segnaposto {
  border-bottom: 1px dashed rgba(0,0,0,.4);
}

.c0 p:first-child {
  margin-top: 0;
}

.c0 p:last-child {
  margin-bottom: 0;
}

.c0 img {
  max-width: 100%;
}

.c0 table {
  border-collapse: collapse;
  border-spacing: 0;
}

.c0 table th {
  background: rgb(231,236,236);
}

.c0 table td,
.c0 table th {
  padding: 0.62em;
  border: 1px solid rgb(231,236,236);
  vertical-align: middle;
}

.c0 .numero-esercizio {
  float: right;
  background-color: rgb(209,209,209);
  color: rgb(255,255,255);
  padding: 3px 6px;
  border-radius: 3px;
  text-align: right;
}

.c0 .editor {
  border: 1px dashed rgb(251,91,111);
}

.c0 .MathJax {
  font-size: 0.9em;
}

.c0 table th {
  background: rgb(231,236,236);
}

.c0 table td,
.c0 table th {
  padding: 0.62em;
  border: 1px solid rgb(231,236,236);
  vertical-align: middle;
}

.c0 .numero-esercizio {
  float: right;
  background-color: rgb(209,209,209);
  color: rgb(255,255,255);
  padding: 3px 6px;
  border-radius: 3px;
  text-align: right;
}

@media print {
  .c0 {
    border: none;
  }
}

@media print {
  .c0 .opzioni {
    background: none;
  }
}

<Div
  className="c0"
  color="red"
  theme={
    Object {
      "brand": "#00BBEF",
    }
  }
/>
`;

exports[`<HtmlBlock /> should render its css w/ props note 1`] = `
.c0 {
  color: #00BBEF;
  background-color: #FFFDDE;
  border-top: 1px solid rgb(231,236,236);
  padding: 16px;
}

.c0 .segnaposto-admin {
  display: none;
}

.c0 .consegna__titolo {
  display: block;
  padding-bottom: 20px;
  font-weight: 600;
}

.c0 .consegna__contenuto {
  color: #000;
}

.c0 .opzioni {
  margin-top: 10px;
  padding: 10px;
  background-color: rgba(96,141,212,0.1);
  border: 1px solid rgba(35,56,89,.1);
  border-radius: 2px;
}

.c0 .opzioni__titolo {
  display: block;
  padding-bottom: 10px;
}

.c0 .opzioni__contenuto {
  color: #000;
  display: inline-block;
  font-style: italic;
  padding-left: 6px;
}

.c0 .opzioni__contenuto .radiocheck-wrap {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 0;
}

.c0 .opzioni .radio,
.c0 .opzioni .checkbox {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  margin-right: 10px;
  padding-top: 3px;
}

.c0 .opzioni .radio:after,
.c0 .opzioni .checkbox:after {
  content: "";
  display: block;
  width: 14px;
  height: 14px;
  border: 1px solid #00BBEF;
  background: white;
}

.c0 .opzioni .checkbox:after {
  border-radius: 2px;
}

.c0 .opzioni .radio:after {
  border-radius: 100px;
}

.c0 .consegna__contenuto--esercizio {
  margin-top: 20px;
}

.c0 .completa {
  line-height: 2em;
}

.c0 .completa:after {
  content: "........................................";
  font-weight: 400;
}

.c0 .completa.full--width {
  display: block;
  height: 0;
  border-bottom: 1px dashed;
  margin: 2em 0;
}

.c0 .completa.full--width:after {
  content: "";
}

.c0 .completa.full--width:last-child {
  margin-bottom: 1em;
}

.c0 .mytestitaliano .contesto {
  border-bottom: 1px dotted;
}

.c0 .consegna__blocchi {
  margin-top: 10px;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 4px;
  padding: 0 4px;
  border-radius: 2px;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco span {
  padding: 4px 0;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco.analizza {
  border: 1px solid rgba(0,0,0,.4);
}

.c0 .consegna__blocchi .consegna__blocchi-blocco.analizza .segnaposto {
  border-bottom: 1px dashed rgba(0,0,0,.4);
}

.c0 p:first-child {
  margin-top: 0;
}

.c0 p:last-child {
  margin-bottom: 0;
}

.c0 img {
  max-width: 100%;
}

.c0 table {
  border-collapse: collapse;
  border-spacing: 0;
}

.c0 table th {
  background: rgb(231,236,236);
}

.c0 table td,
.c0 table th {
  padding: 0.62em;
  border: 1px solid rgb(231,236,236);
  vertical-align: middle;
}

.c0 .numero-esercizio {
  float: right;
  background-color: rgb(209,209,209);
  color: rgb(255,255,255);
  padding: 3px 6px;
  border-radius: 3px;
  text-align: right;
}

.c0 .editor {
  border: 1px dashed rgb(251,91,111);
}

.c0 .MathJax {
  font-size: 0.9em;
}

.c0 table th {
  background: rgb(231,236,236);
}

.c0 table td,
.c0 table th {
  padding: 0.62em;
  border: 1px solid rgb(231,236,236);
  vertical-align: middle;
}

.c0 .numero-esercizio {
  float: right;
  background-color: rgb(209,209,209);
  color: rgb(255,255,255);
  padding: 3px 6px;
  border-radius: 3px;
  text-align: right;
}

@media print {
  .c0 {
    border: none;
  }
}

@media print {
  .c0 .opzioni {
    background: none;
  }
}

<Div
  className="c0"
  note={true}
  theme={
    Object {
      "brand": "#00BBEF",
    }
  }
/>
`;

exports[`<HtmlBlockFooter /> should render a Div 1`] = `
.c0 {
  color: rgb(51,51,51);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 16px;
  border-top: 1px solid rgb(231,236,236);
}

@media print {
  .c0 {
    border: none;
  }
}

<Div
  className="c0"
/>
`;

exports[`<HtmlInjector /> should not render HtmlBlock 1`] = `
<Div>
  <HtmlInjector__Solution
    dangerouslySetInnerHTML={
      Object {
        "__html": "soluzione",
      }
    }
  />
  <HtmlInjector__HtmlBlockFooter>
    <span>
      16 ESERCIZI
    </span>
    <span>
      25/01/2018
    </span>
  </HtmlInjector__HtmlBlockFooter>
</Div>
`;

exports[`<HtmlInjector /> should not render HtmlBlockFooter w/o data e esercizi 1`] = `
<Div>
  <HtmlInjector__HtmlBlock
    dangerouslySetInnerHTML={
      Object {
        "__html": "html",
      }
    }
    note={true}
    theme={
      Object {
        "brand": "#00BBEF",
      }
    }
  />
  <HtmlInjector__Solution
    dangerouslySetInnerHTML={
      Object {
        "__html": "soluzione",
      }
    }
  />
</Div>
`;

exports[`<HtmlInjector /> should not render MathJax when editor.watch flag is set 1`] = `
<Div>
  <Div
    border="1px dashed rgb(251, 91, 111)"
    padding="16px"
  >
    <CKEditor
      activeClass="activeConsegnaEditor"
      config={Object {}}
      content=""
      events={
        Object {
          "change": undefined,
        }
      }
      isScriptLoaded={false}
      scriptUrl="https://cdn.ckeditor.com/4.6.2/standard/ckeditor.js"
    />
  </Div>
  <HtmlInjector__Solution>
    <CKEditor
      activeClass="activeSoluzioneEditor"
      config={Object {}}
      content=""
      events={
        Object {
          "change": undefined,
        }
      }
      isScriptLoaded={false}
      scriptUrl="https://cdn.ckeditor.com/4.6.2/standard/ckeditor.js"
    />
  </HtmlInjector__Solution>
  <HtmlInjector__HtmlBlockFooter>
    <span>
      16 ESERCIZI
    </span>
    <span>
      25/01/2018
    </span>
  </HtmlInjector__HtmlBlockFooter>
</Div>
`;

exports[`<HtmlInjector /> should not render MathJax when hasLatex flag is set 1`] = `
<Div>
  <HtmlInjector__HtmlBlock
    note={true}
    overflowy="auto"
    theme={
      Object {
        "brand": "#00BBEF",
      }
    }
  >
    <_default
      math="html"
    />
  </HtmlInjector__HtmlBlock>
  <HtmlInjector__Solution>
    <_default
      math="soluzione"
    />
  </HtmlInjector__Solution>
  <HtmlInjector__HtmlBlockFooter>
    <span>
      16 ESERCIZI
    </span>
    <span>
      25/01/2018
    </span>
  </HtmlInjector__HtmlBlockFooter>
</Div>
`;

exports[`<HtmlInjector /> should not render Solution 1`] = `
<Div>
  <HtmlInjector__HtmlBlock
    dangerouslySetInnerHTML={
      Object {
        "__html": "html",
      }
    }
    note={true}
    theme={
      Object {
        "brand": "#00BBEF",
      }
    }
  />
  <HtmlInjector__HtmlBlockFooter>
    <span>
      16 ESERCIZI
    </span>
    <span>
      25/01/2018
    </span>
  </HtmlInjector__HtmlBlockFooter>
</Div>
`;

exports[`<HtmlInjector /> should render HtmlBlockFooter w/ data when esercizi is 0 1`] = `
<Div>
  <HtmlInjector__HtmlBlock
    dangerouslySetInnerHTML={
      Object {
        "__html": "html",
      }
    }
    note={true}
    theme={
      Object {
        "brand": "#00BBEF",
      }
    }
  />
  <HtmlInjector__Solution
    dangerouslySetInnerHTML={
      Object {
        "__html": "soluzione",
      }
    }
  />
  <HtmlInjector__HtmlBlockFooter>
    <span>
      0 ESERCIZI
    </span>
    <span>
      25/01/2018
    </span>
  </HtmlInjector__HtmlBlockFooter>
</Div>
`;

exports[`<HtmlInjector /> should render HtmlBlockFooter w/ data when esercizi is 1 1`] = `
<Div>
  <HtmlInjector__HtmlBlock
    dangerouslySetInnerHTML={
      Object {
        "__html": "html",
      }
    }
    note={true}
    theme={
      Object {
        "brand": "#00BBEF",
      }
    }
  />
  <HtmlInjector__Solution
    dangerouslySetInnerHTML={
      Object {
        "__html": "soluzione",
      }
    }
  />
  <HtmlInjector__HtmlBlockFooter>
    <span>
      1 ESERCIZIO
    </span>
    <span>
      25/01/2018
    </span>
  </HtmlInjector__HtmlBlockFooter>
</Div>
`;

exports[`<HtmlInjector /> should render HtmlBlockFooter w/ only data 1`] = `
<Div>
  <HtmlInjector__HtmlBlock
    dangerouslySetInnerHTML={
      Object {
        "__html": "html",
      }
    }
    note={true}
    theme={
      Object {
        "brand": "#00BBEF",
      }
    }
  />
  <HtmlInjector__Solution
    dangerouslySetInnerHTML={
      Object {
        "__html": "soluzione",
      }
    }
  />
  <HtmlInjector__HtmlBlockFooter>
    <span>
      25/01/2018
    </span>
  </HtmlInjector__HtmlBlockFooter>
</Div>
`;

exports[`<HtmlInjector /> should render a Div 1`] = `
<Div>
  <HtmlInjector__HtmlBlock
    dangerouslySetInnerHTML={
      Object {
        "__html": "html",
      }
    }
    note={true}
    theme={
      Object {
        "brand": "#00BBEF",
      }
    }
  />
  <HtmlInjector__Solution
    dangerouslySetInnerHTML={
      Object {
        "__html": "soluzione",
      }
    }
  />
  <HtmlInjector__HtmlBlockFooter>
    <span>
      16 ESERCIZI
    </span>
    <span>
      25/01/2018
    </span>
  </HtmlInjector__HtmlBlockFooter>
</Div>
`;

exports[`<Solution /> should render a HtmlBlock 1`] = `
.c0 {
  color: #0a6b06;
  background-color: #d0ffc8;
}

<HtmlInjector__HtmlBlock
  className="c0"
  theme={
    Object {
      "brand": "#00BBEF",
    }
  }
/>
`;

exports[`<Solution /> should render its css w/ props bgcolor 1`] = `
.c0 {
  color: #0a6b06;
  background-color: green;
}

<HtmlInjector__HtmlBlock
  bgcolor="green"
  className="c0"
  theme={
    Object {
      "brand": "#00BBEF",
    }
  }
/>
`;

exports[`<Solution /> should render its css w/ props color 1`] = `
.c0 {
  color: red;
  background-color: #d0ffc8;
}

<HtmlInjector__HtmlBlock
  className="c0"
  color="red"
  theme={
    Object {
      "brand": "#00BBEF",
    }
  }
/>
`;
